src/main/perl/FileWriter.pm

```diff
@@ -301,10 +303,22 @@ sub close
                 $msg = 'would have been';
                 $self->debug(1, "File $filename with NoAction=1");
             } else {
+                my $parent_dir = dirname($filename);
+                # Pass NoAction here, as it keeps track of the NoAction value during initialisation and/or keeps_state
+                my $cafpath = CAF::Path::mkcafpath(log => *$self->{LOG}, NoAction => $options->{noaction});
+                # only create the directory if it didn't exist yet
+                # if not, this would for the directory mode on existing directories
```

Comment:
Sorry I don't understand this comment. Did you mean "if not, this would change the directory mode on existing directories"?